src/symbolic_dimensions.jl

```diff
@@ -68,7 +90,36 @@ function SymbolicDimensions{R}(; kws...) where {R}
     V = R[tryrationalize(R, kws[i]) for i in p]
     return SymbolicDimensions{R}(permute!(I, p), V)
 end
-(::Type{<:SymbolicDimensions})(::Type{R}; kws...) where {R} = SymbolicDimensions{R}(; kws...)
+function SymbolicDimensionsSingleton{R}(s::Symbol) where {R}
+    i = get(ALL_MAPPING, s, INDEX_TYPE(0))
+    iszero(i) && error("$s is not available as a symbol in `SymbolicDimensionsSingleton`. Symbols available: $(ALL_SYMBOLS).")
+    return SymbolicDimensionsSingleton(i, R)
+end
+
+# Traits:
+dimension_names(::Type{<:AbstractSymbolicDimensions}) = ALL_SYMBOLS
+Base.propertynames(::AbstractSymbolicDimensions) = ALL_SYMBOLS
+Base.getindex(d::AbstractSymbolicDimensions, k::Symbol) = getproperty(d, k)
+constructorof(::Type{<:SymbolicDimensions}) = SymbolicDimensions
+constructorof(::Type{<:SymbolicDimensionsSingleton{R}}) where {R} = SymbolicDimensionsSingleton{R}
+with_type_parameters(::Type{<:SymbolicDimensions}, ::Type{R}) where {R} = SymbolicDimensions{R}
+with_type_parameters(::Type{<:SymbolicDimensionsSingleton}, ::Type{R}) where {R} = SymbolicDimensionsSingleton{R}
+nzdims(d::SymbolicDimensions) = getfield(d, :nzdims)
+nzdims(d::SymbolicDimensionsSingleton) = [getfield(d, :dim)]
```

Comment:
Could this be a tuple for `SymbolicDimensionsSingleton`? These allocations seem undesirable.
```suggestion
nzdims(d::SymbolicDimensionsSingleton) = (getfield(d, :dim),)
```

Reply:
Maybe it would be easier to just define all methods for the `...Singleton` type directly, without going through something like `nzdims`.